### PR TITLE
Darken the color of links within click tooltips to improve color contrast and readability

### LINKED ISSUE #9559: Links in tooltips are too light color, hard to read / click
### Describe the bug

The link in this tooltip looks too light.

Image

### To Reproduce

1. Go to the deposits list page, ie wp-admin > Payments > Deposits (or Payouts)
2. Click on the deposit schedule tooltip like the screenshot above.

### Actual behavior

Links' color is too light making it almost unreadable.

### Additional context

Discovered while working on https://github.com/Automattic/woocommerce-payments/pull/9528.

### PR DESCRIPTION
**Pull request by Jinksi**
Fixes #9559

#### Changes proposed in this Pull Request

This PR fixes the color contrast of links within tooltips to improve readability.

**Before/after**
<div style="display: flex;">

</div>

#### Testing instructions

1. Go to the following pages and ensure that tooltips are rendered with appropriate color contrast.
	1. Payments → Deposits list schedule notice
	2. Payments → Overview (account balances, deposit card)
	3. Payments → Settings (colors should remain unchanged)
	4. Payments → Transactions list (Future Deposit tooltip)

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️) **Not currently, but we should consider visual regression testing**
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9571` or branch name `fix/9559-tooltip-link-color-contrast` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/9559-tooltip-link-color-contrast) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 097daae8f725eca20b6b894abd0f056dbce8abd0
- Build time: 2024-10-17 23:11:15 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/components/tooltip/style.scss [hunk 1]
```diff
@@ -72,16 +74,21 @@
...
-			// Specific styles for the click tooltip variant.
+			// Click tooltips have a light background and dark text.
 			position: relative;
 			padding: 12px;
 			color: $gray-900;
 			text-align: left;
 			border-radius: 2px;
 			border: 1px solid $gray-400;
-			background: $white;
+			background-color: $white;
```
Comment (Jinksi) [inline client/components/tooltip/style.scss:84]: Minor change to use consistent naming to default tooltip.

File: client/components/tooltip/style.scss [hunk 2]
```diff
@@ -72,16 +74,21 @@
...
 			box-shadow: 0 2px 6px 0 rgba( 0, 0, 0, 0.05 );
 
+			// Links inside click tooltips should have a dark color against the light background.
+			a {
+				color: var( --wp-admin-theme-color, $gutenberg-blue );
```
Comment (Jinksi) [inline client/components/tooltip/style.scss:89]: Using the original colors from before the inadvertent change in #9273

Comment (shendy-a8c) [reply client/components/tooltip/style.scss:89]:
```suggestion
				color: $gutenberg-blue;
```

I wonder if we should just set the link color statically like so, without involving the `--wp-admin-theme-color` variable into the equation.

[We set the tooltip background to be `$white` statically](https://github.com/Automattic/woocommerce-payments/blob/ed68ce24c15d667e0ed50ab68a3e629b57df3541/client/components/tooltip/style.scss#L84). If the link color is dynamic, depending on the value of that `--wp-admin-theme-color` variable, there is a possibility where the link color is not readable. For example, if somehow that variable is set to `$white` (maybe for dark mode, color theming, or some other reason), we'll have white link on top of a white background.

What do you think?

**Review by shendy-a8c — APPROVED** [review]
Nice work! I confirm that this change fixes the problem but I have some feedback to consider.

File: client/components/tooltip/style.scss (hunk 2 above)
Comment (Jinksi) [reply client/components/tooltip/style.scss:89]: Yep, it looks especially weird when we have another element within the tooltip coloured blue, and the links coloured by the theme colour.

image

Comment (Jinksi) [reply client/components/tooltip/style.scss:89]: The entire payments overview page is a mismatch of theme colours and hard-coded colours. I'll open a new issue to rectify this.

image

Comment (Jinksi) [reply client/components/tooltip/style.scss:89]: I think it should be sufficient to support these default themes – and for colour consistency, we should prefer these theme colours.

For this PR, I say we stick with the theme colour (previous behaviour before bug was introduced) and make a clear decision on which colour to use in the subsequent colour consistency issue.

image

<details>
<summary>Here's how the tooltip links look with each</summary>

image
image
image
image
image
image
image
image
image
image


</details>

Comment (Jinksi) [reply client/components/tooltip/style.scss:89]: But yes, other WP themes extend these admin colour themes, which we'll need to keep in mind.

For example, Tsubaki, which was the default WooExpress theme, defaults to this "Classic Dark" theme but adds other options:

image
image
image

Comment (Jinksi) [reply client/components/tooltip/style.scss:89]: #9584 created to address the inconsistent colours when a custom admin theme is enabled

**Pull request merged**